### PR TITLE
Use dumb-init for Clair entrypoint

### PR DESCRIPTION
**Pull request by bseb**
If Clair attempts to update its vulnerability database and network connectivity is interrupted zombie git-http process are created and never reaped. Dumb-init was created specifically to address this zombie process reaping issue in Docker. This solution has been tested and is working for me. 

https://github.com/Yelp/dumb-init

### CODE REVIEW

**Comment by bseb** [comment]
This change has been merged by Coreos/clair as well  https://github.com/coreos/clair/pull/477#pullrequestreview-69589350

**Comment by reasonerjt** [comment]
Thanks @bseb, we need other review process to merge it as it introduce external binary, I'll keep it open and follow up. 
Could you check the build break?

**Comment by bseb** [comment]
It appears to me the failed CI builds are unrelated to this PR, please examine the output

**Comment by bseb** [comment]
@reasonerjt Any movement on this PR?

**Comment by reasonerjt** [comment]
@bseb 
Thanks for asking.

Sorry for the late response, there's a parallel work to use non-root user to run clair container, 
would you mind updating the code to introduce the dumb-init to latest code?
Let me know if you are too busy to handle this, I'll make the update.

**Comment by bseb** [comment]
@reasonerjt Ive updated the Dockerfile to reflect changes to master branch

**Comment by bseb** [comment]
@reasonerjt Are you saying you want a copy of dumb-init binary in this repository?

File: make/photon/clair/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,10 @@
 FROM vmware/photon:1.0
 
+RUN tdnf install -y git bzr rpm xz \
```
Comment (reasonerjt) [inline make/photon/clair/Dockerfile]: please remove duplicate lines of `tdnf install`

File: make/photon/clair/Dockerfile [hunk 2]
```diff
@@ -9,11 +14,19 @@ RUN tdnf distro-sync -y \
     && useradd --no-log-init -m -r -g 10000 -u 10000 clair
 COPY clair /clair2.0.1/
 COPY docker-entrypoint.sh /docker-entrypoint.sh
+
 VOLUME /config
+
 EXPOSE 6060 6061
+
+RUN chmod u+x /clair2.0.1/clair
```
Comment (reasonerjt) [inline make/photon/clair/Dockerfile]: Remove the dup line.

File: make/photon/clair/Dockerfile [hunk 3]
```diff
@@ -9,11 +14,19 @@ RUN tdnf distro-sync -y \
...
 RUN chown -R 10000:10000 /clair2.0.1 \
     && chmod u+x /clair2.0.1/clair \
     && chmod u+x /docker-entrypoint.sh
+
 HEALTHCHECK --interval=30s --timeout=10s --retries=3 CMD curl -sS 127.0.0.1:6061/health || exit 1
+
 USER clair
-ENTRYPOINT ["/docker-entrypoint.sh"]
+
+ENTRYPOINT ["/usr/local/bin/dumb-init", "--", "/clair2.0.1/clair"]
```
Comment (reasonerjt) [inline make/photon/clair/Dockerfile]: Could you update the entrypoint.sh to use dumb-init?

File: make/photon/clair/Dockerfile [hunk 4]
```diff
@@ -9,11 +14,19 @@ RUN tdnf distro-sync -y \
     && useradd --no-log-init -m -r -g 10000 -u 10000 clair
 COPY clair /clair2.0.1/
 COPY docker-entrypoint.sh /docker-entrypoint.sh
```
Comment (reasonerjt) [inline make/photon/clair/Dockerfile]: use COPY to copy the dumb-init.

File: make/photon/clair/Dockerfile [hunk 5]
```diff
@@ -9,11 +14,19 @@ RUN tdnf distro-sync -y \
...
+
 VOLUME /config
+
 EXPOSE 6060 6061
+
+RUN chmod u+x /clair2.0.1/clair
+
 RUN chown -R 10000:10000 /clair2.0.1 \
     && chmod u+x /clair2.0.1/clair \
```
Comment (reasonerjt) [inline make/photon/clair/Dockerfile]: chmod dumb-init here.

**Comment by reasonerjt** [comment]
@bseb 
Yes, please upload the binary to the repo, instead of downloading it from internet.

Please also see my comments in the PR, there are a few dup lines.

**Comment by bseb** [comment]
@reasonerjt  Requested changes complete

Side note: In my opinion using docker-entrypoint.sh obfuscates what the entrypoint is actually doing. Consider making the entrypoint more explicit

**Pull request merged**